code/modules/mob/living/carbon/human/say.dm

```diff
@@ -13,14 +20,20 @@
 /mob/living/carbon/human/GetVoice()
 	if(istype(wear_mask, /obj/item/clothing/mask/chameleon))
 		var/obj/item/clothing/mask/chameleon/V = wear_mask
-		if(V.vchange && wear_id)
+		if(V.voice_change && wear_id)
 			var/obj/item/card/id/idcard = wear_id.GetID()
 			if(istype(idcard) && idcard.electric)
 				return idcard.registered_name
 			else
 				return real_name
 		else
 			return real_name
+	if(istype(wear_mask, /obj/item/clothing/mask/gas/old/modulator))
+		var/obj/item/clothing/mask/gas/old/modulator/V = wear_mask
+		if(V.voice_change)
+			return "Unknown"
+		else
+			return real_name
```

Comment:
This would be better as a proc on mask, which can transform the name.
```
/obj/item/clothing/mask/proc/get_name(original_name)
  return original_name
```
And then add specific logic on the masks themselves.